File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/comment/direct/Java/Python/s831281137_commented.py
import sys

def main():
    N, W = map(int, sys.stdin.readline().split())
    v = [0] * (N + 1)
    w = [0] * (N + 1)
    for i in range(1, N + 1):
        a, b = map(int, sys.stdin.readline().split())
        v[i] = a
        w[i] = b
    sumV = [[0] * (W + 1) for _ in range(N + 1)]
    for i in range(1, N + 1):
        for i_sum in range(0, w[i]):
            sumV[i][i_sum] = sumV[i-1][i_sum]
        for i_sum in range(w[i], W + 1):
            option1 = sumV[i-1][i_sum]
            option2 = sumV[i-1][i_sum - w[i]] + v[i]
            sumV[i][i_sum] = max(option1, option2)
    print(sumV[N][W])

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/cot/Java/Go/s831281137_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	sc := bufio.NewScanner(os.Stdin)
	sc.Split(bufio.ScanWords)

	N := nextInt(sc)
	W := nextInt(sc)

	v := make([]int, N+1)
	w := make([]int, N+1)

	for i := 1; i <= N; i++ {
		v[i] = nextInt(sc)
		w[i] = nextInt(sc)
	}

	sumV := make([][]int, N+1)
	for i := range sumV {
		sumV[i] = make([]int, W+1)
	}

	for i := 1; i <= N; i++ {
		for iSum := 0; iSum < w[i]; iSum++ {
			sumV[i][iSum] = sumV[i-1][iSum]
		}

		for iSum := w[i]; iSum <= W; iSum++ {
			if sumV[i-1][iSum-w[i]]+v[i] > sumV[i-1][iSum] {
				sumV[i][iSum] = sumV[i-1][iSum-w[i]] + v[i]
			} else {
				sumV[i][iSum] = sumV[i-1][iSum]
			}
		}
	}

	fmt.Println(sumV[N][W])
}

func nextInt(sc *bufio.Scanner) int {
	sc.Scan()
	i, e := strconv.Atoi(sc.Text())
	if e != nil {
		panic(e)
	}
	return i
}

// End of Code
